commitlint.config.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {
+    extends: ['@commitlint/config-conventional'],
+
+    helpUrl: 'https://github.com/camunda/zeebe/blob/main/CONTRIBUTING.md/#commit-message-guidelines',
+
+    // Rules are made up by a name and a configuration array. The configuration array contains:
+    // - Level [0..2]: 0 disables the rule. For 1 it will be considered a warning for 2 an error.
+    // - Applicable always|never: never inverts the rule.
+    // - Value: value to use for this rule.
+    rules: {
+        // error rules
+        'body-max-line-length': [2, 'always', 120],
+        'header-max-length': [2, 'always', 120],
+        'type-case': [2, 'always', 'lower-case'],
+        'type-empty': [2, 'never'],
+        'type-enum': [
+            2,
+            'always',
+            [
+                `build`, // Changes that affect the build system (e.g. Maven, Docker, etc)
+                `ci`, // Changes to our CI configuration files and scripts (e.g. Jenkins, Bors, etc)
+                `deps`, // A change to the external dependencies (was already used by Dependabot)
+                `docs`, //  A change to the documentation
+                `feat`, // A new feature (both internal or user-facing)
+                `fix`, // A bug fix (both internal or user-facing)
+                'merge', // Merges one or multiple commits
+                `perf`, // A code change that improves performance
+                `refactor`, // A code change that does not change the behavior
+                'revert', // Reverts any change
+                `style`, // A change to align the code with our style guide
+                `test`, // Adding missing tests or correcting existing tests
+            ],
+        ],
+
+        // warning rules
+
+
+        // disabled rules
+        'body-full-stop': [0, 'never', '.'],
+        'body-leading-blank': [0, 'always'],
+        'body-empty': [0, 'never'],
+        'body-max-length': [0, 'always', Infinity],
+        'body-min-length': [0, 'always', 0],
+        'body-case': [0, 'always', 'lower-case'],
+        'footer-leading-blank': [0, 'always'],
+        'footer-empty': [0, 'never'],
+        'footer-max-length': [0, 'always', Infinity],
+        'footer-max-line-length': [0, 'always', 100],
+        'footer-min-length': [0, 'always', 0],
+        'header-case': [0, 'always', 'lower-case'],
+        'header-full-stop': [0, 'never', '.'],
+        'header-min-length': [0, 'always', 0],
+        'references-empty': [0, 'never'],
+        'scope-enum': [0, 'always', []],
+        'scope-case': [0, 'always', 'lower-case'],
+        'scope-empty': [0, 'never'],
```

Comment:
👍 Valid point. Scope is optional in both the guidelines and this ruleset:

Rules starting with `0` are disabled; they are not checked. Let's look at some examples:

- `'scope-empty': [2, 'never']`: 2 means error and the rule should never match, so scope is not allowed to be empty
- `'scope-empty': [1, 'never']`: same as above, but only a warning - 1 means warning
- `'scope-empty': [2, 'always']`: the rule should always match - scope should always be empty
- `'scope-empty': [0, 'always']`: the rule is disabled (0) - it does not matter whether the rule should match always or never because it's not checked

Reply:
Thanks Nico 😃 a bit confusing Syntax 😅

Reply:
Haha absolutely agree, it took me some time to understand it as well. That's why I grouped the rules as `error`, `warning`, and `disabled` in the hope this would clarify it 😄

Reply:
Maybe we add a comment to show the enum, like error(2), warning(1), none(0).

Reply:
😅 like this? https://github.com/camunda/zeebe/pull/12175/files/b9aff1e2c444631b2cbf6354cd90185f2ccbe28f#diff-2b1ae4316086c0ebb213227790133a792300e1cef03b0afc067b23d3399ea5caR7